HisashiburiDana.Application/Services/AuthenticationService.cs

```diff
@@ -178,5 +184,185 @@ public async Task<GeneralResponseWrapper<string>> ReIssueAccessToken(RefreshToke
             return response.BuildSuccessResponse(newAccessToken);
 
         }
+
+        public async Task<GeneralResponseWrapper<bool?>> SendCodeToEmail(string email)
+        {
+            _logger.LogInformation($"Login Request Arrived ---{email}");
+            GeneralResponseWrapper<bool?> response = new(_logger);
+
+            try
+            {
+                //get user from data
+                var user = await _unitOfWork.UserRepo.Get("Email", ScanOperator.Equal, email);
+                if (user == null)
+                {
+                    List<string> errors = new()
+                {
+                    "incorrect email address, kindly input the email attached to this account"
+                };
+                    return response.BuildFailureResponse(errors);
+                }
+
+                //* Generate random code and send to email address
+                var code = GenerateRandomCode();
+
+                //* Send code to email
+                var status = _emailService.SendPasswordResetEmail(email, code);
+                if (!status)
+                {
+                    List<string> errors = new()
+                    {
+                     "an error occurred while sending passcode to email"
+                    };
+                    return response.BuildFailureResponse(errors);
+                }
+
+                // Saving the code in the user's record in the database
+                user.PassCode = code;
```

Comment:
Since we are using DDD. A domain should only be changed inside itself by itself. I suggest you create a static method inside the user class to handle the PassCode update